src/overlay005/field_control.c

```diff
@@ -905,14 +905,14 @@ static BOOL Field_UpdatePoison(FieldSystem *fieldSystem)
         return FALSE;
     }
 
-    switch (sub_02054B04(party, MapHeader_GetMapLabelTextID(fieldSystem->location->mapId))) {
+    switch (Pokemon_DoPoisonDamage(party, MapHeader_GetMapLabelTextID(fieldSystem->location->mapId))) {
     case 0:
```

Comment:
**suggestion:** It would be good to add descriptive constants for these cases (and the corresponding return values in `Pokemon_DoPoisonDamage()`.